lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.js

```diff
@@ -0,0 +1,151 @@
+
+'use strict';
+
+const _ = require('lodash');
+const BbPromise = require('bluebird');
+
+// NOTE --> Keep this file in sync with ../stage.js
+
+// NOTE: This code was written since there are problem setting up dedicated CloudFormation
+// Stage resource (see https://github.com/serverless/serverless/pull/5692#issuecomment-467849311 for more information).
+
+module.exports = {
+  updateStage() {
+    // this array is used to gather all the patch operations we need to
+    // perform on the stage
+    this.apiGatewayStagePatchOperations = [];
+    this.apiGatewayStageState = null;
+
+    return BbPromise.bind(this)
+      .then(getAccountId)
+      .then(getRestApiId)
+      .then(getStage)
+      .then(handleTracing)
+      .then(handleLogs)
+      .then(handleTags)
+      .then(applyUpdates);
+  },
+};
+
+function getAccountId() {
+  return this.provider.getAccountId().then((id) => this.accountId = id); // eslint-disable-line
+}
+
+function getRestApiId() {
+  return this.provider.request('APIGateway', 'getRestApis', { limit: 5000 })
+    .then((res) => {
+      if (res.items.length) {
+        const filteredRestApis = res.items.filter((api) =>
+          api.name === `${this.options.stage}-${this.serverless.service.service}`);
+        if (filteredRestApis.length) {
+          const restApi = filteredRestApis.shift();
+          return restApi.id;
+        }
+      }
+      return null;
+    })
+    .then((restApiId) => {
+      this.apiGatewayRestApiId = restApiId;
+    });
+}
+
+function getStage() {
+  const restApiId = this.apiGatewayRestApiId;
+
+  return this.provider.request('APIGateway', 'getStage', {
+    restApiId,
+    stageName: this.options.stage,
+  }).then((res) => {
+    this.apiGatewayStageState = res;
+  });
+}
+
+function handleTracing() {
+  const tracing = this.serverless.service.provider.tracing;
+  const tracingEnabled = !_.isEmpty(tracing) && tracing.apiGateway;
+
+  let operation = { op: 'replace', path: '/tracingEnabled', value: 'false' };
+  if (tracingEnabled) {
+    operation = { op: 'replace', path: '/tracingEnabled', value: 'true' };
+  }
+  this.apiGatewayStagePatchOperations.push(operation);
+}
+
+function handleLogs() {
+  const provider = this.serverless.service.provider;
+  const logs = provider.logs && provider.logs.restApi;
+  const ops = this.apiGatewayStagePatchOperations;
+
+  let dataTrace = { op: 'replace', path: '/*/*/logging/dataTrace', value: 'false' };
+  let logLevel = { op: 'replace', path: '/*/*/logging/loglevel', value: 'OFF' };
+
+  let operations = [dataTrace, logLevel];
+
+  if (logs) {
+    const service = this.serverless.service.service;
+    const stage = this.options.stage;
+    const region = this.options.region;
+    const logGroupName = `/aws/api-gateway/${service}-${stage}`;
+
+    // eslint-disable-next-line max-len
+    const destinationArn = { op: 'replace', path: '/accessLogSettings/destinationArn', value: `arn:aws:logs:${region}:${this.accountId}:log-group:${logGroupName}` };
```

Comment:
Why not make it multi-line (to avoid eslint disabling) ?

Reply:
Yes, I tried that but AWS expects this to be a 1 LOC string.

Reply:
hmm... but how AWS depends on code structure of this module (?) Technically I think it doesn't make difference whether this object literal is written as above, or as below (which doesn't require eslint disabling)

```javascript
const destinationArn = {
  op: 'replace',
  path: '/accessLogSettings/destinationArn',
  value: `arn:aws:logs:${region}:${this.accountId}:log-group:${logGroupName}`,
};
```

What do I miss?